pkg/symbolizer/symbolizer.go

```diff
@@ -84,36 +85,100 @@ func (s *Symbolizer) SymbolizePprof(ctx context.Context, profile *googlev1.Profi
 		stringMap[str] = int64(i)
 	}
 
-	var allSymbolizedLocs []symbolizedLocation
+	// Symbolize all mappings concurrently
+	allSymbolizedLocs, err := s.symbolizeMappingsConcurrently(ctx, profile, locationsByMapping)
+	if err != nil {
+		return fmt.Errorf("symbolizing mappings: %w", err)
+	}
+
+	s.updateAllSymbolsInProfile(profile, allSymbolizedLocs, stringMap)
+
+	return nil
+}
+
+// symbolizeMappingsConcurrently symbolizes multiple mappings concurrently with a concurrency limit.
+func (s *Symbolizer) symbolizeMappingsConcurrently(
+	ctx context.Context,
+	profile *googlev1.Profile,
+	locationsByMapping map[uint64][]*googlev1.Location,
+) ([]symbolizedLocation, error) {
+	// Concurrency limit to avoid overwhelming debuginfod server
+	const maxConcurrency = 10
```

Comment:
Could be nice to have this as a config value so it can be tweaked without a recompile

Reply:
Done it in 0e360590c453e3f3167f7f4636262ced6363eace